plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -823,6 +860,88 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
         transaction = null;
     }
 
+    @Override
+    public void executeTableExecute(ConnectorSession session, ConnectorTableExecuteHandle tableExecuteHandle)
+    {
+        IcebergTableExecuteHandle executeHandle = (IcebergTableExecuteHandle) tableExecuteHandle;
+        switch (executeHandle.getProcedureId()) {
+            case EXPIRE_SNAPSHOTS:
+                executeExpireSnapshots(session, executeHandle);
+                return;
+            default:
+                throw new IllegalArgumentException("Unknown procedure '" + executeHandle.getProcedureId() + "'");
+        }
+    }
+
+    private void executeExpireSnapshots(ConnectorSession session, IcebergTableExecuteHandle executeHandle)
+    {
+        IcebergExpireSnapshotsHandle expireSnapshotsHandle = (IcebergExpireSnapshotsHandle) executeHandle.getProcedureHandle();
+
+        Table table = catalog.loadTable(session, executeHandle.getSchemaTableName());
+        int tableVersion = ((BaseTable) table).operations().current().formatVersion();
+        if (tableVersion > EXPIRE_SNAPSHOTS_MAX_SUPPORTED_TABLE_VERSION) {
+            throw new TrinoException(NOT_SUPPORTED, format("EXPIRE_SNAPSHOTS is not supported for iceberg table version > %d. " +
+                            "Table %s version is %s.",
+                    EXPIRE_SNAPSHOTS_MAX_SUPPORTED_TABLE_VERSION,
+                    executeHandle.getSchemaTableName(),
+                    tableVersion));
+        }
+        Map<String, String> properties = table.properties();
+        if (properties.containsKey(WRITE_LOCATION_PROVIDER_IMPL)) {
+            throw new TrinoException(NOT_SUPPORTED, "Table " + executeHandle.getSchemaTableName() + " specifies " + properties.get(WRITE_LOCATION_PROVIDER_IMPL) +
+                    " as a location provider. Writing to Iceberg tables with custom location provider is not supported.");
+        }
```

Comment:
Why exactly? Can you please explain with code-level comment?

Reply:
> Sorry if I asked this already, but do you have a test with v2 tables that have delete marker files?
> 
> I thought you'd need to include `fileScanTask.deletes()` when doing both procedures

You did ask that but I think my answer may not be correct. Could you please advise how to create such a test? As far as I know currently deletes work only on partitions